internal/catalogmetadata/client/client.go

```diff
@@ -70,93 +83,81 @@ func (c *Client) Bundles(ctx context.Context) ([]*catalogmetadata.Bundle, error)
 		if err != nil {
-			return nil, err
-		}
-
-		allBundles = append(allBundles, bundles...)
-	}
-
-	return allBundles, nil
-}
-
-func PopulateExtraFields(catalogName string, channels []*catalogmetadata.Channel, bundles []*catalogmetadata.Bundle, deprecations []*catalogmetadata.Deprecation) ([]*catalogmetadata.Bundle, error) {
-	bundlesMap := map[string]*catalogmetadata.Bundle{}
-	for i := range bundles {
-		bundleKey := fmt.Sprintf("%s-%s", bundles[i].Package, bundles[i].Name)
-		bundlesMap[bundleKey] = bundles[i]
-
-		bundles[i].CatalogName = catalogName
-	}
-
-	for _, ch := range channels {
-		for _, chEntry := range ch.Entries {
-			bundleKey := fmt.Sprintf("%s-%s", ch.Package, chEntry.Name)
-			bundle, ok := bundlesMap[bundleKey]
-			if !ok {
-				return nil, fmt.Errorf("bundle %q not found in catalog %q (package %q, channel %q)", chEntry.Name, catalogName, ch.Package, ch.Name)
-			}
-
-			bundle.InChannels = append(bundle.InChannels, ch)
+			return nil, fmt.Errorf("error processing response: %s", err)
 		}
-	}
-
-	// According to https://docs.google.com/document/d/1EzefSzoGZL2ipBt-eCQwqqNwlpOIt7wuwjG6_8ZCi5s/edit?usp=sharing
-	// the olm.deprecations FBC object is only valid when either 0 or 1 instances exist
-	// for any given package
-	deprecationMap := make(map[string]*catalogmetadata.Deprecation, len(deprecations))
-	for _, deprecation := range deprecations {
-		deprecationMap[deprecation.Package] = deprecation
-	}
 
-	for i := range bundles {
-		if dep, ok := deprecationMap[bundles[i].Package]; ok {
-			for _, entry := range dep.Entries {
+		for _, deprecation := range deprecations {
+			for _, entry := range deprecation.Entries {
 				switch entry.Reference.Schema {
 				case declcfg.SchemaPackage:
-					bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
+					for _, pkg := range packages {
+						if pkg.Name == deprecation.Package {
+							pkg.Deprecation = &declcfg.DeprecationEntry{
+								Reference: entry.Reference,
+								Message:   entry.Message,
+							}
+						}
+					}
 				case declcfg.SchemaChannel:
-					for _, ch := range bundles[i].InChannels {
-						if ch.Name == entry.Reference.Name {
-							bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
-							break
+					for _, channel := range channels {
+						if channel.Package == deprecation.Package && channel.Name == entry.Reference.Name {
+							channel.Deprecation = &declcfg.DeprecationEntry{
+								Reference: entry.Reference,
+								Message:   entry.Message,
+							}
 						}
 					}
 				case declcfg.SchemaBundle:
-					if bundles[i].Name == entry.Reference.Name {
-						bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
+					for _, bundle := range bundles {
+						if bundle.Package == deprecation.Package && bundle.Name == entry.Reference.Name {
+							bundle.Deprecation = &declcfg.DeprecationEntry{
+								Reference: entry.Reference,
+								Message:   entry.Message,
+							}
```

Comment:
I similarly recall the conversation as keeping the wrapper as simple as possible if one is necessary (at the very least we need one for that catalog name since the `declcfg.*` representations don't store catalog information). I recall @joelanford mentioning adding the deprecation entry to the wrapper as well in previous conversations so I took that approach as it was one less set of information to deal with after the fact. I'm not opposed to either of the approaches though so this is more or less a "let's pick one and I'll implement it" situation. 

The benefit of this approach is that there is no lookup cost after the set of packages, channels, and bundles are returned but there are definitely other solutions where the lookup cost is negligible that I'm open to exploring.

Reply:
I don't remember the context of my suggestion of putting the deprecation entry in the wrapper, but I think I'd suggest separating it out and keeping the structure similar to the FBC structure.